tests/farm/LiquidityMining.js

```diff
@@ -682,59 +810,540 @@ contract("LiquidityMining", (accounts) => {
+                        .div(HUNDRED_PERCENT),
+                }
+            );
+        });
+
+        it("should be able to claim reward using wrapper", async () => {
+            let depositTx = await liquidityMining.deposit(token1.address, amount, ZERO_ADDRESS, {
+                from: account1,
+            });
+            let depositBlockNumber = new BN(depositTx.receipt.blockNumber);
+            await SOVToken.transfer(liquidityMining.address, new BN(1000));
+
+            let tx = await wrapper.claimReward(token1.address, { from: account1 });
+
+            let poolInfo = await liquidityMining.getPoolInfo(token1.address);
+            let latestBlockNumber = new BN(tx.receipt.blockNumber);
+            checkPoolInfo(
+                poolInfo,
+                token1.address,
+                allocationPoint,
+                latestBlockNumber,
+                new BN(-1)
+            );
+
+            await checkUserPoolTokens(account1, token1, amount, amount, new BN(0));
+            await checkUserReward(account1, token1, depositBlockNumber, latestBlockNumber);
+
+            // withdrawAndStakeTokensFrom was invoked
+            let unlockedBalance = await lockedSOV.getUnlockedBalance(account1);
+            let lockedBalance = await lockedSOV.getLockedBalance(account1);
+            expect(unlockedBalance).bignumber.equal(new BN(0));
+            expect(lockedBalance).bignumber.equal(new BN(0));
+        });
+
+        it("fails if token pool token not found", async () => {
+            await expectRevert(
+                liquidityMining.claimReward(account1, ZERO_ADDRESS, { from: account1 }),
+                "Pool token not found"
+            );
+        });
+    });
+
+    describe("claimRewardFromAllPools", () => {
+        let allocationPoint = new BN(1);
+        let amount = new BN(1000);
+
+        beforeEach(async () => {
+            await deploymentAndInit();
+            await liquidityMining.add(token1.address, allocationPoint, false);
+            await liquidityMining.add(token2.address, allocationPoint, false);
+            await mineBlocks(1);
+
+            await token1.mint(account1, amount);
+            await token1.approve(liquidityMining.address, amount, { from: account1 });
+            await token2.mint(account1, amount);
+            await token2.approve(liquidityMining.address, amount, { from: account1 });
+        });
+
+        it("shouldn't be able to claim reward (will not be claimed without SOV tokens)", async () => {
+            await liquidityMining.deposit(token1.address, amount, ZERO_ADDRESS, {
+                from: account1,
+            });
+
+            await expectRevert(
+                liquidityMining.claimRewardFromAllPools(ZERO_ADDRESS, { from: account1 }),
+                "Claiming reward failed"
+            );
+        });
+
+        it("should be able to claim all rewards (will be claimed with SOV tokens) with 10% of unlockedImmediatelyPercent", async () => {
```

Comment:
pls add comments from the ticket requirement checkpoints (SOV-1983) to the respective tests. otherwise it is hard to check the tests.

Reply:
updated